superset/models/reports.py

```diff
@@ -0,0 +1,172 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+# pylint: disable=line-too-long,unused-argument,ungrouped-imports
+"""A collection of ORM sqlalchemy models for Superset"""
+import enum
+
+from flask_appbuilder import Model
+from sqlalchemy import (
+    Boolean,
+    Column,
+    DateTime,
+    Float,
+    ForeignKey,
+    Integer,
+    String,
+    Table,
+    Text,
+)
+from sqlalchemy.orm import relationship
+from sqlalchemy.schema import UniqueConstraint
+
+from superset.extensions import security_manager
+from superset.models.core import Database
+from superset.models.dashboard import Dashboard
+from superset.models.helpers import AuditMixinNullable
+from superset.models.slice import Slice
+
+metadata = Model.metadata  # pylint: disable=no-member
+
+
+class ReportScheduleType(str, enum.Enum):
+    ALERT = "Alert"
+    REPORT = "Report"
+
+
+class ReportRecipientType(str, enum.Enum):
+    EMAIL = "Email"
+    SLACK = "Slack"
+
+
+class ReportLogState(str, enum.Enum):
+    SUCCESS = "Success"
+    ERROR = "Error"
+
+
+class ReportEmailFormat(str, enum.Enum):
+    VISUALIZATION = "Visualization"
+    DATA = "Raw data"
+
+
+report_schedule_user = Table(
+    "report_schedule_user",
+    metadata,
+    Column("id", Integer, primary_key=True),
+    Column("user_id", Integer, ForeignKey("ab_user.id"), nullable=False),
+    Column(
+        "report_schedule_id", Integer, ForeignKey("report_schedule.id"), nullable=False
+    ),
+    UniqueConstraint("user_id", "report_schedule_id"),
+)
+
+
+class ReportSchedule(Model, AuditMixinNullable):
```

Comment:
I think we're missing a `description = Column(text)` column that explains what the report/alert is about (as internal documentation purposes), and maybe a `context_markdown = Column(text)` that would be provided as context when the alert/report is sent, and would be part of the email that gets sent. Both can be left empty.

This could potentially be a single field too that serves both purposes. @willbarrett what do you think?

Reply:
Yes, let's add `description` and `context_markdown`, but punt the context work for a later iteration.

Reply:
Done